### PR TITLE
[WIP] FAQ.md

### PR DESCRIPTION
**Pull request by dmp42**
@stevvooe:

First rough draft.

What do you think generally? Should this be just one FAQ? Or should we have a generic one, and another one more specifically focused on upgrade/compat issues?

### CODE REVIEW

File: docs/FAQ.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+Support for the hub is handled directly by Docker, over here: https://support.docker.com
+
+## What if I don't want to rely on the Docker Hub?
+
+Then you should run and operate your own registry. 
+
+While this solution will require more work from you (like running and operating any kind of service does), it will give you more control over your images, and more configurability.
+
+## Are there different registries implementations available?
```
Comment (thaJeztah) [inline docs/FAQ.md]: s/registries/registry/

File: docs/FAQ.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+Yes, there exist at least two different.
+
+First, the "old, legacy registry":
+
+ * implemented in python
+ * source code is here: https://github.com/docker/docker-registry
+ * supports the v1 protocol
```
Comment (thaJeztah) [inline docs/FAQ.md]: link to the protocol?

File: docs/FAQ.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+## What if I don't want to rely on the Docker Hub?
+
+Then you should run and operate your own registry. 
+
+While this solution will require more work from you (like running and operating any kind of service does), it will give you more control over your images, and more configurability.
+
+## Are there different registries implementations available?
+
+Yes, there exist at least two different.
```
Comment (thaJeztah) [inline docs/FAQ.md]: Think this sentence needs a fix

File: docs/FAQ.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+ * implements a new, "v2" protocol
+ * is were most of the development is happening, and the right place for new and upcoming ideas 
+
+
+## Which one should I use then?
+
+
+### Rule of thumb
+
+... and strong recommandation: please use the latest Docker version available (>=1.6), and the new, golang registry.
```
Comment (thaJeztah) [inline docs/FAQ.md]: s/recommandation/recommendation/

File: docs/FAQ.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+To achieve that, you will have to run both a legacy registry, and a new registry, both behind a nginx proxy doing the routing.
+
+TODO: details about that
+
+### Yelling: "I DONT WANT TO CHANGE! WHY DO I HAVE TO CHANGE? YOU MORONS!"
+
+Good news for you: "YOU DONT HAVE TO CHANGE ANYTHING".
+
+You can keep running your old registry with newer docker engine versions. The docker engine is retro-compatible and will happily use whatever registry you are running.
```
Comment (thaJeztah) [inline docs/FAQ.md]: docker (not referring to the CLI binary) should start with a capital (Docker); here, and various other places above

File: docs/FAQ.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+ * is currently in maintenance mode: only critical security fixes are being worked on
+
+Second, the "new registry":
+
+ * implemented in go
+ * source code is here: https://github.com/docker/distribution
+ * implements a new, "v2" protocol
+ * is were most of the development is happening, and the right place for new and upcoming ideas 
+
+
```
Comment (thaJeztah) [inline docs/FAQ.md]: Should this section have a question; "I don't like those; can I create my *own* implementation?"

I doubt many people want to, but mention that it's possible if they comply with the protocol (again, link to the specs/protocol)

File: docs/FAQ.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+## So, does the Docker Hub contain a registry?
+
+Yes.
+
+That "Hub registry" is operated by Docker.
+
+It is free to use for public content, and offers paid-for private storage as well.
+
+Support for the hub is handled directly by Docker, over here: https://support.docker.com
+
```
Comment (thaJeztah) [inline docs/FAQ.md]: A mention what the differences are between Docker Hub and the open source project? (Web interface not included, etc)

File: docs/FAQ.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+## What if I don't want to rely on the Docker Hub?
+
+Then you should run and operate your own registry. 
+
+While this solution will require more work from you (like running and operating any kind of service does), it will give you more control over your images, and more configurability.
+
```
Comment (thaJeztah) [inline docs/FAQ.md]: "Can I host/mirror official images on my own registry?"
"Can I use my own registry as the *default* registry, in stead of Docker Hub?"

**Comment by thaJeztah** [comment]
Added some random blurbs while reading, may come up with more. Thanks!

**Comment by stevvooe** [comment]
Does this belong in docs or repository root?

**Comment by thaJeztah** [comment]
The docker docs also has a FAQ; https://docs.docker.com/faq/ so (I think) it would make sense to have a FAQ for this project as well.

It can still be linked from the README on github

File: docs/FAQ.md (hunk 7 above)
Comment (dmp42) [inline docs/FAQ.md]: I don't want to overload registry documentation with details about other products (like the Hub). If people want to learn about the hub they should go there (furthermore, if they landed here, they are already pretty deep in).

One of the defect of Docker documentation at large IMHO is how cluttered and unwieldy it is.

File: docs/FAQ.md (hunk 2 above)
Comment (dmp42) [inline docs/FAQ.md]: Do you know where it is? :)

File: docs/FAQ.md (hunk 6 above)
Comment (dmp42) [inline docs/FAQ.md]: I think that's out of scope of this document.
Also... sure they can implement their own - it's open-source.

**Comment by dmp42** [comment]
Thanks for the feedback @thaJeztah 
In a shell: I would like to keep this document relatively small and to the point, focused on the most critical questions.

File: docs/FAQ.md (hunk 7 above)
Comment (thaJeztah) [inline docs/FAQ.md]: Agreed, it is in need of a serious overhaul (especially now the other projects are included)

I do see questions regularly regarding "where's the source for Docker Hub", that's why I was thinking some short note could be meaningful.

But..these are all just quick "blurbs" from my side, so no problem

**Comment by stevvooe** [comment]
@dmp42 Do you want me to leave this on the Registry/2.0 milestone? This needs a little work before we merge.

**Comment by dmp42** [comment]
I'll push to get it done tomorrow.

**Comment by stevvooe** [comment]
@dmp42 Ping!

**Comment by dmp42** [comment]
Meh.

**Comment by stevvooe** [comment]
@dmp42 Do we want to push this forward?

**Comment by dmp42** [comment]
Calling this of. Documentation has been vastly enhanced, people are moving away from v1, and the "migrator" seems to do a good job for these who wanna migrate.

**Pull request closed without merging**